### PR TITLE
feat: cast send,call, estimate print revert reason(data)

### PR DESCRIPTION
**Pull request by cuiweixie**
## Solution
closes https://github.com/foundry-rs/foundry/issues/8606  closes https://github.com/foundry-rs/foundry/issues/8603

### CODE REVIEW

File: crates/cast/bin/tx.rs [hunk 1]
```diff
@@ -270,7 +271,29 @@ where
...
+                                error_payload,
+                                error_payload.data.as_ref()
+                                    .map_or_else(String::new, |err_data| err_data.get().trim_matches('"').to_string())
+                            );
+                            return Err(ErrReport::msg(err_str))
+                        }
+                        _ =>{
+                            return Err(e.into())
+                        }
+                    };
```
Comment (mattsse) [inline crates/cast/bin/tx.rs:294]: I think we try using

`error_payload.as_decoded_error::<GenericContractError>()` here?

Comment (cuiweixie) [reply crates/cast/bin/tx.rs]: I try, but it seems error_payload can not decode as GenericContractError.

Comment (cuiweixie) [reply crates/cast/bin/tx.rs]:
```  
          Err(e) => {
                match e {
                    RpcError::ErrorResp(error_payload) => {
                        return Err(match error_payload.as_decoded_error::<GenericContractError>(false) {
                            Some(contract_error) => ErrReport::msg(contract_error.to_string()),
                            None => ErrReport::msg("No contract error"),
                        });
                    }
                    _ => {
                        return Err(e.into())
                    }
                };
            }
```

result in No contract error

Comment (cuiweixie) [reply crates/cast/bin/tx.rs]: It seems needs the abi.

**Comment by cuiweixie** [comment]
I am not sure that if the git push -f has overwrite your request change by accident.

**Comment by klkvr** [comment]
I am wondering if this should be updated in alloy instead?

In ethers all fields of error payload were formatted: 
https://github.com/gakonst/ethers-rs/blob/34ed9e372e66235aed7074bc3f5c14922b139242/ethers-providers/src/rpc/transports/common.rs#L73

vs in alloy:
https://github.com/alloy-rs/alloy/blob/68c8bfc2b786e6e460475b28e0f2453bbcf90e55/crates/json-rpc/src/response/error.rs#L86

**Comment by cuiweixie** [comment]
Repo alloy has merged the pr.  
when will foundry include the [revert data](https://github.com/alloy-rs/alloy/commit/512c8ad256708a3173485b9f76c59c1cf4373b87).

**Pull request closed without merging**